test/unit/org/apache/cassandra/index/sai/cql/AnalyzerTest.java

```diff
@@ -29,6 +29,19 @@
 
 public class AnalyzerTest extends SAITester
 {
+    @Test
+    public void testEmpty()
+    {
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, v text)");
+        createIndex("CREATE CUSTOM INDEX ON %s(v) USING 'StorageAttachedIndex' WITH OPTIONS = {" +
+                    "'index_analyzer': 'standard'};");
+        execute("INSERT INTO %s (k, v) VALUES (1, 'apple orange')");
+        flush();
+        execute("INSERT INTO %s (k, v) VALUES (1, '')");
+        flush();
+        compact();
```

Comment:
Nit: Maybe we could add a final query checking query results, just in case:
```suggestion
        compact();
        assertRows(execute("SELECT * FROM %s WHERE v = 'apple'"));
```